### PR TITLE
autodoc: Change dictionary sort order

### PR DESCRIPTION
**Pull request by khwilliamson**
This makes this more in line with Data::Dumper sorting.

upper/lower case continues to not matter, and numbers continue to come after letters, so that `ckWARN2()` comes after plain `ckWARN()`.

It changes non-leading underscores to come before letters, so that `ck_warner` comes before `ckWARN`.

And it changes so leading underscores come after non-leading, so that `aMY_CXT` and `aMY_CXT_` come before
`_aMY_CXT`

### CODE REVIEW

**Comment by jkeenan** [comment]
> This makes this more in line with Data::Dumper sorting.
> 

To review this pull request, I built blead through `make test_prep` and did likewise with a branch built from your p.r. (rebased on blead).  I then compared the respective `pod/perlapi.pod` generated files.

> It changes numbers to come after letters, so that ckWARN2() comes after ckWARN().
>

Grepping the two generated files (as described below), I couldn't find any `=item` entries that illustrate this objective.  How would I search the generated files to find examples of this objective?
 
> It changes non-leading underscores to come before letters, so that ABC_DEF comes before ABCDEF, as the former is likely to be seen as two words, and ABC should come before ABCD.

If I grepped for `=item` entries in the two files ...
```
ack '^=item C<[^>]*>' pod/perlapi.pod
```
... then took a `diff` of the two greps, then I observed:
``` 
663,667d662
< =item C<LONGDBLINFBYTES>
< =item C<LONGDBLMANTBITS>
< =item C<LONGDBLNANBYTES>
< =item C<LONG_DOUBLEKIND>
< =item C<LONG_DOUBLESIZE>
672a668,672
> =item C<LONG_DOUBLEKIND>
> =item C<LONG_DOUBLESIZE>
> =item C<LONGDBLINFBYTES>
> =item C<LONGDBLMANTBITS>
> =item C<LONGDBLNANBYTES>
```
... which appears to meet your objective.

> 
> And it changes so leading underscores come after non-leading, so aTHX comes before _aTHX

Using the search procedure described above, I found in blead:
```
 4078 =head1 Concurrency
 4079 
 4080 =over 4
 4081 
 4082 =item C<aTHX>
 4083 
 4084 =item C<aTHX_>
 4085 
 4086 Described in L<perlguts>.
 4087 
 4088 =back
 4089 
```
... while in the branch I found:
```
 4052 =head1 Concurrency
 4053 
 4054 =over 4
 4055 
 4056 =item C<aTHX>
 4057 
 4058 =item C<aTHX_>
 4059 
 4060 Described in L<perlguts>.
 4061 
 4062 =back
 4063 
```
So there was no change in `aTHX` versus `aTHX_`.  Can you provide an example of where this change took effect?

**Comment by khwilliamson** [comment]
Sorry for my glib description. I forgot that this retained the existing sort order of numbers where the come after letters.  I changed the commit message to give real examples of the things that did change

File: autodoc.pl [hunk 1]
```diff
@@ -2162,45 +2162,40 @@ sub dictionary_order {
...
-    my $a_only_word = uc($a =~ s/$mod_W//gr);
-    my $b_only_word = uc($b =~ s/$mod_W//gr);
-
-    # And not initial nor interior underscores nor digits (by squeezing them
-    # out)
-    my $a_stripped = $a_only_word =~ s/ (*atomic:[$ud]+) (*pla: $mod_w ) //grxx;
-    my $b_stripped = $b_only_word =~ s/ (*atomic:[$ud]+) (*pla: $mod_w ) //grxx;
+    # Strip out \W.
+    my $a_stripped = $a =~ s/$mod_W//gr;
+    my $b_stripped = $b =~ s/$mod_W//gr;
```
Comment (tonycoz) [inline autodoc.pl]: All this duplicated code between `$a` and `$b` could go into a function.

Comment (khwilliamson) [reply autodoc.pl]: changed

**Review by tonycoz — APPROVED** [review]

**Pull request merged**